### PR TITLE
Prune unmanaged resources

### PR DESCRIPTION
**Pull request by andreasgerstmayr**
Prune objects which exist in the cluster but are not managed by the operator anymore. For example, when the Jaeger Query Ingress is enabled and later disabled, the Ingress object should be removed from the cluster.

Resolves: #259

### CODE REVIEW

**Comment by andreasgerstmayr** [comment]
Could we increase the covscan limit a bit, or do we really want to test every error path? :neutral_face:
(I'd need to find a way to fake a kubernetes API error when deleting an object for example)

File: controllers/tempo/tempostack_controller_test.go [hunk 1]
```diff
@@ -419,3 +421,74 @@ func TestTLSEnable(t *testing.T) {
...
+	}
+	req := ctrl.Request{
+		NamespacedName: nsn,
+	}
+	reconcileResult, err := reconciler.Reconcile(context.Background(), req)
+	require.NoError(t, err)
+	assert.Equal(t, false, reconcileResult.Requeue)
+
+	// Verify Ingress is created
+	ingressNsn := types.NamespacedName{Name: "tempo-prune-ingress-test-query-frontend", Namespace: "default"}
```
Comment (iblancasa) [inline controllers/tempo/tempostack_controller_test.go:473]: Maybe, you should do the same for routes (just to increase the coverage)

**Review by iblancasa — APPROVED** [review]

File: controllers/tempo/tempostack_controller_test.go (hunk 1 above)
Comment (andreasgerstmayr) [reply controllers/tempo/tempostack_controller_test.go:473]: The tempostack_controller tests use a Kubernetes control plane (`KUBEBUILDER_ASSETS` env var in the Makefile).
When I create a unit test with a Route object I get a panic when calling `obj.GetName()` on the Route object.
I already expected that it won't work, but I'm surprised that it panics :smiling_face_with_tear:

Comment (andreasgerstmayr) [reply controllers/tempo/tempostack_controller_test.go:473]: So afaics we can only test Routes in e2e tests, when the OpenShift router is installed.

**Pull request merged**